### DISCUSSION TITLE
How to use cordova plugins like camera or file opener without the use of jetifier?

### DISCUSSION THREAD

**Discussion by vineetha-v** [ionic-team/capacitor#4460] [Q&A]
I am trying to create a custom plugin which makes use of camera and file functionalities.  I want this plugin to work with both Cordova and Capacitor (without using jetifier), can someone suggest a better way to get this done real quick.

**Comment by jcesarmobile** [comment]
Best thing is to use AndroidX dependencies instead of the deprecated Android Support libraries.
cordova-android 9 already supports AndroidX, but it's disabled by default. Users can enable it by adding `<preference name="AndroidXEnabled" value="true" />` to their `config.xml`

But sadly, if users have some plugin that has not moved to AndroidX, they'll have to use jetifier or cordova-plugin-androidx-adapter or some other thing to patch those plugins. That's not really up to you, you can't know what plugins they use.

**Reply by EinfachHans** [reply]
With Cordova-Android 10, AndroidX will be default, so i hope the most used plugins (like file-opener2) will update their code then 🤷🏼‍♂️
